filebeat/scripts/generator/fields/main.go

```diff
@@ -95,17 +96,18 @@ func newFieldYml(name, typeName string, noDoc bool) *fieldYml {
 func newField(lp string) field {
 	lp = lp[1 : len(lp)-1]
```

Comment:
This code is fine as long as we use it just as generator on the dev side. But as recently we started to move some code also in the Beats binary we should probably be more careful with validation checks. What happens here in case an empty string is passed to it?

Reply:
This function never gets an empty string as a param, because it parses the strings returned by `FindAllString` which does not match for empty lines. But just to be sure, if some tries to use it incorrectly in the future, I added additional error checking.

Reply:
I think the key here is we can't make assumption about future usage of this method as you stated so checks should be in place to make sure the method never panics.

Seems the checks didn't make it in yet? If I run `newField("")` the code would panic?